mpi4jax/flush.py

```diff
@@ -7,3 +7,8 @@ def flush():
     for device in jax.devices("cpu"):
```

Comment:
Can't this just be 

```python
devices = [*jax.devices("cpu"), *jax.devices("gpu")]
for device in devices:
    ...
```

?

Reply:
`jax.devices("gpu")` fails if you don't have a gpu.

You should probe what are the available devices, but that is an implementation detail subject to change in jaxlib and you also get extra devices (like interpreters...) that don't mean anything.

Reply:
I think I would prefer

```python
devices = jax.devices("cpu")

try:
    devices.extend(jax.devices("gpu"))
except RuntimeError:
    pass
```

then. Also prevents the double flush on CPU.